layouts/partials/article/title.html

```diff
@@ -2,17 +2,75 @@
 {{ $articleLink := ($.Scratch.Get "articleLink") }}
 
 {{ if (eq .Parent.Title .Title) }}
-{{/*  No title - For when the main title is the same as the first article  */}}
+{{/* No title - For when the main title is the same as the first article */}}
 {{ else }}
     <div class="article-title article-actions" data-align="center-left">
-        {{ if .Data.Pages }}
-            <h1> {{ .Title }}</h1>
+
+        {{/* Calculate depth by counting parents, ignoring the global site title (`$.Site.Title`) to avoid an extra heading level */}}
+        {{ $depth := 0 }}
+        {{ $currentPage := . }}
+
+        {{ with $currentPage.Parent }}
+            {{ if ne .Title $.Site.Title }}
+                {{ $depth = add $depth 1 }}
+            {{ end }}
+            
+            {{ with .Parent }}
+                {{ if ne .Title $.Site.Title }}
+                    {{ $depth = add $depth 1 }}
+                {{ end }}
+                
+                {{ with .Parent }}
```

Comment:
Can we rather use recursion here and just pass the depth value and then return when no more parents

Reply:
You'll need to create a new partial